src/passwordcheck.js

```diff
@@ -27,12 +27,26 @@ define([
       PASSWORD_TOO_SHORT: 'PASSWORD_TOO_SHORT'
     };
 
-    function isStrongEnough(password) {
-      // Check for passwords that at least contain a number and an alphabet,
-      // or if alphabets, then at least (minLength + 4) characters long
-      var regexString = '((?=.*[a-zA-Z])(?=.*[0-9])[A-Za-z0-9!@#$%^&*()_+ ]{' + minLength + ',' + (minLength + 4) + '})|([A-Za-z0-9!@#$%^&*()_+ ]{' + (minLength + 4) + ',})';
-      var regex = new RegExp(regexString);
-      return regex.test(password);
+    function isAllLetters(password) {
+      return /^[a-zA-Z]+$/.test(password);
+    }
+
+    function isAllNumbers(password) {
+      return /^[0-9]+$/.test(password);
+    }
+
+    function isWeakPassword(password) {
+      // Check for passwords that consist of only numbers
+      // or only alphabets, and are of length less than (minlength + 4)
+      if (password.length >= (minLength + 4)) {
```

Comment:
@TDA - I simplified this logic slightly so the regexp's could be simpler and easier to understand. First, if the password.length, we don't worry about it here, it'll be checked by the bloom filter automatically. If the password.length < minLength + 4, then check whether it is all letters or all numbers.